go-controller/pkg/ovn/pods.go

```diff
@@ -205,7 +205,7 @@ func (oc *DefaultNetworkController) addLogicalPort(pod *kapi.Pod) (err error) {
 		if err != nil {
 			return err
 		}
-	} else if config.Gateway.DisableSNATMultipleGWs {
+	} else if config.Gateway.DisableSNATMultipleGWs && !util.PodCompleted(pod) {
```

Comment:
not sure this is necessary. I think it is protected by the retry framework:

```
		cache.ResourceEventHandlerFuncs{
			AddFunc: func(obj interface{}) {
				r.ResourceHandler.RecordAddEvent(obj)

				key, err := GetResourceKey(obj)
				if err != nil {
					klog.Errorf("Upon add event: %v", err)
					return
				}
				klog.V(5).Infof("Add event received for %s %s", r.ResourceHandler.ObjType, key)

				r.DoWithLock(key, func(key string) {
					// This only applies to pod watchers (pods + dynamic network policy handlers watching pods):
					// if ovnkube-master is restarted, it will get all the add events with completed pods
					if r.ResourceHandler.IsObjectInTerminalState(obj) {
						r.processObjectInTerminalState(obj, key, resourceEventAdd)
						return
					}
```

Reply:
For update it is handled here:
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/retry/obj_retry.go#L541

and here:
https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/retry/obj_retry.go#L588

The add/update events are what trigger something to be added to retry. But since they both check for terminal state they would never be added to retry as something to add/update, but instead added as something to delete.